doc/rfc007-foreign-calls.md

```diff
@@ -0,0 +1,220 @@
+# RFC007: Design space of foreign calls in Quint
+
+| Revision | Date       | Author           |
+| :------- | :--------- | :--------------- |
+| 1        | 2023-04-05 | Igor Konnov      |
+
+## 1. Summary
+
+In this RFC, we discuss potential solutions for integrating the Quint simulator
+with other execution environments. This would enrich the simulator with
+computations that cannot be expressed in Quint itself, e.g., rich string
+manipulation and parsing. Additionally, it would give us another path for more
+interactive integration of the Quint simulator with systems under test.
+
+## 2. Context
+
+Quint is a specification language with the following features that are important
+in the context of this RFC:
+
+ - Quint has a relatively small standard library, and
+ - Quint has formal semantics, thanks to its transpilation to TLA+.
+
+Thanks to these features, Quint is an easier target for static analysis and
+verification tools than general-purpose programming languages. At the same time,
+these nice features are somewhat in conflict with the expressivity and usability
+of the language, from the systems engineer's point of view. Even though Quint is
+obviously Turing-complete (for example, quantifiers over integers and integer
+arithmetic make many properties undecidable), this does not mean that all of the
+typical engineering tasks are easy to do in Quint.
+
+In the following, we introduce several examples that motivate the need for this
+RFC.
+
+### Example 1: Strings
```

Comment:
For strings, there was another request for string concatenation on [Zulip](https://informal-systems.zulipchat.com/#narrow/stream/265309-apalache/topic/String.20concatenation), but for Apalache rather than Quint.